crates/rattler_installs_packages/src/wheel_builder_frontend.py

```diff
@@ -0,0 +1,109 @@
+import sys
+from sys import exit
+from pathlib import Path
+from importlib import import_module
+from json import loads
+from types import ModuleType
+import json
+
+################################################################
+# Begin janky attempt to workaround
+#   https://github.com/pypa/setuptools/issues/3786
+################################################################
+
+try:
+    import sysconfig
+    import distutils.sysconfig
+
+
+    def get_python_inc(plat_specific=0, prefix=None):
+        if plat_specific:
+            return sysconfig.get_path("platinclude")
+        else:
+            return sysconfig.get_path("include")
+
+
+    distutils.sysconfig.get_python_inc = get_python_inc
+except ImportError:
+    # we ignore missing distutils (was removed in Python 3.12)
+    pass
+
+################################################################
+# End janky workaround
+################################################################
+
+def get_backend_from_entry_point(entrypoint: str) -> ModuleType:
+    # https://packaging.python.org/en/latest/specifications/entry-points/
+    modname, qualname_separator, qualname = entrypoint.partition(":")
+    backend = import_module(modname)
+    if qualname_separator:
+        for attr in qualname.split("."):
+            backend = getattr(backend, attr)
+            if backend is None:
+                raise AttributeError(f"Attribute '{attr}' not found in '{modname}'")
+
+    return backend
+
+def get_requires_for_build_wheel(backend: ModuleType, work_dir: Path) -> [str]:
+    """
+    Return an list of requirements. This is only necessary if we do not
+    have a pyproject.toml file.
+    """
+    f = getattr(backend, "get_requires_for_build_wheel")
+    if f is None:
+        result = []
+    else:
+        result = f()
+
+    j = json.dumps(result)
+    out_json_file = work_dir / "extra_requirements.json"
+    out_json_file.write_text(j)
```

Comment:
Could we not just use stdout directly for this?

Reply:
Unfortunately that doesn't work because some of the `setup.py` scripts also write some garbage to the `stdout`...